### DISCUSSION TITLE
Different number of search results for every page load

### DISCUSSION THREAD

**Discussion by thany** [community/community#40344] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Please load this url:
https://github.com/search?q=There+are+multiple+modules+with+names+that+only+differ+in+casing.&type=code

And then reload it a few times. You'll see the number of results be different each time.

Should remain the same, shouldn't it?

If this is intended (and I dare not ask why, if so), what's the logic exactly?

**Comment by Akash1134** [comment]
Hi @thany , This happens because of an internal search template that works differently with each search logic. This [discussion](https://github.com/microsoft-search/pnp-modern-search/issues/1413) should give  you more details of how it works. Thanks!

**Reply by thany** [reply]
Yes, so basically the results are coming from a server that, by itself, produces inconsistent resultsets... Surely, _surely_, that's a bug?

**Comment by look** [comment, marked as answer]
The results are affected by timeouts in the backend search engine. The idea is that it's better to give you some results than time out completely.

That said, if you haven't [joined our beta](https://github.com/features/code-search) for the new code search, I recommend doing so. It is faster and more reliable.

**Reply by thany** [reply]
When you say "by timeouts in the backend search engine" my brain goes "Fix it then, please!"

**Reply by thany** [reply]
Can I also ask please, how can a beta be more reliable than a production version? If it really is, then deploy it!

**Reply by look** [reply]
The beta is powered by a new custom search engine we wrote for code search. We are working on scaling it up so we can let everyone use it.
